dandi/metadata.py

```diff
@@ -115,18 +72,6 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
-    # Somewhat less fragile search than previous proposals,
```

Comment:
ok, so in effect `get_metadata` becomes `get_nwb_or_dandiset_metadata`, and that is what breaks `ls` as @jwodder described in https://github.com/dandi/dandi-cli/pull/1076#issuecomment-1205304196 . For now IMHO it is ok to just disable `ls` test on BIDS but we need to clear up this "generic" `get_metadata` aspect. For that let's discuss in a separate issue https://github.com/dandi/dandi-cli/issues/1096